src/Console/MergeContacts.php

```diff
@@ -0,0 +1,165 @@
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Console;
+
+use Friendica\Core\L10n;
+use Friendica\Database\Database;
+
+/**
+ * tool to find and merge duplicated contact entries.
+ */
+class MergeContacts extends \Asika\SimpleConsole\Console
+{
+	protected $helpOptions = ['h', 'help', '?'];
+
+	/**
+	 * @var $dba Database
+	 */
+	private $dba;
+
+	/**
+	 * @var L10n
+	 */
+	private $l10n;
+
+	protected function getHelp()
+	{
+		$help = <<<HELP
+console mergecontacts - Merge duplicated contact entries
+Synopsis
+	bin/console mergecontacts
+
+Description
+	bin/console mergecontacts
+		Remove duplicated contact entries
+
+Options
+	-h|--help|-? Show help information
+	-e|--execute Execute the dropping
+HELP;
+		return $help;
+	}
+
+	public function __construct(Database $dba, L10n $l10n, array $argv = null)
+	{
+		parent::__construct($argv);
+
+		$this->dba  = $dba;
+		$this->l10n = $l10n;
+	}
+
+	protected function doExecute()
+	{
+		$duplicates = $this->dba->p("SELECT COUNT(*) AS `total`, `uri-id`, MAX(`url`) AS `url` FROM `contact` WHERE `uid` = 0 GROUP BY `uri-id` HAVING total > 1");
+		while ($duplicate = $this->dba->fetch($duplicates)) {
+			$this->out($this->l10n->t('%d %s, %d duplicates.', $duplicate['uri-id'], $duplicate['url'], $duplicate['total']));
+			if ($this->getOption(['e', 'execute'], false)) {
+				$this->mergeContacts($duplicate['uri-id']);
+			}
+		}
+		return 0;
+	}
+
+	private function mergeContacts(int $uriid)
+	{
+		$first = $this->dba->selectFirst('contact', ['id', 'nurl', 'url'], ["`uri-id` = ? AND `nurl` != ? AND `url` != ?", $uriid, '', ''], ['order' => ['id']]);
+		if (empty($first)) {
+			$this->err($this->l10n->t('No valid first countact found for uri-id %d.', $uriid));
+			return;
+		}
+		$this->out($first['url']);
+
+		$duplicates = $this->dba->select('contact', ['id', 'nurl', 'url'], ['uri-id' => $uriid]);
+		while ($duplicate = $this->dba->fetch($duplicates)) {
+			if ($first['id'] == $duplicate['id']) {
+				continue;
+			}
+			if ($first['url'] != $duplicate['url']) {
+				$this->err($this->l10n->t('Wrong duplicate found for uri-id %d in %d (url: %s != %s).', $uriid, $duplicate['id'], $first['url'], $duplicate['url']));
+				continue;
+			}
+			if ($first['nurl'] != $duplicate['nurl']) {
```

Comment:
This would trigger even if the duplicate field is empty, is it intended?

Reply:
When the field id empty then something is wrong with that record because that field should be filled. So then we better skip this record.

Reply:
I assume no reply means it's intended?

Reply:
I guess my brain is still sleeping: I don't understand what you wanted to express with your last sentence.

Reply:
It’s all good.